firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/HoneywellABP/Device.cpp

```diff
@@ -10,39 +10,35 @@
  *    Arduino library by ij96 (https://github.com/ij96)
  */
 
-#include "Pufferfish/Driver/I2C/HoneywellABP.h"
+#include "Pufferfish/Driver/I2C/HoneywellABP/Device.h"
 
 #include <array>
 #include <climits>
 
-namespace Pufferfish::Driver::I2C {
+#include "Pufferfish/Util/Endian.h"
+#include "Pufferfish/Util/Ranges.h"
 
-I2CDeviceStatus HoneywellABP::read_sample(ABPSample &sample) {
+namespace Pufferfish::Driver::I2C::HoneywellABP {
+
+I2CDeviceStatus Device::read_sample(ABPSample &sample) {
   std::array<uint8_t, 2> data{{0, 0}};
   I2CDeviceStatus ret = dev_.read(data.data(), data.size());
   if (ret != I2CDeviceStatus::ok) {
     return ret;
   }
 
-  static const uint8_t status_shift = 6;
-  static const size_t bridge_high = 0;
```

Comment:
@ethanjli is this a typo? shouldn't it be 1? and by using `read_bigend` we wont be facing the problem there.

Reply:
It could be a typo, this was implemented by someone else and I'm not familiar with the I2C interface of the ABP sensor so I don't know anything about the bridge data. If it's trying to read multiple bytes of data, read_bigend is probably an appropriate refactor here.